client/webserver/site/src/js/ws.js

```diff
@@ -124,7 +124,11 @@ class MessageSocket {
           reloader()
         }
         forward('open', null, this.handlers)
-        while (this.queue.length) {
+        // Ensure websocket is open when executing queued requests
+        // to avoid infinite request/queue loops.
+        while (this.queue.length &&
+                this.connection &&
+                this.connection.readyState === window.WebSocket.OPEN) {
```

Comment:
Nice detective work!  I think this revealed a deeper bug though.  Adding a console.log here indicates that `readyState` is [2 (CLOSING)](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/readyState), but the [docs for `onopen`](https://developer.mozilla.org/en-US/docs/Web/API/WebSocket/onopen) state "that is called when the WebSocket connection's readyState changes to 1".

Something is fishy with the `conn` and `this.connection` variables.  I've also observed firefox continuing to show several connection failures in the console log after dexc is up and listening, as if there are a back log of attempts and the event handler is stuck on on an old attempt.

If we patch it like this however, I was worried that we'd drop these queued requests, but there also seems to be a quirk where it reloads the markets page and the queued messages are lost anyway.  @buck54321 will have to provide feedback, but you clearly found the source of js burning cpu.

Reply:
I did not notice the conn local to connect(). I believe at the very least I should check the local conn rather than the class level connection. 

Changing it to use the local conn does still prevent the loop.

I see how this patch could result in dropped requests & that probably does need more attention.

Reply:
The latest push I made returns after calling reloader() in conn.onopen(). reloader() reloads the page and triggers another disconnect/connect so I believe all action after reloader() should be obsolete. This also fixes the infinite loop issue.

I've removed the ready state check because as you pointed out that onopen() is called when ready state changes to 1 (OPEN) and therefore my check of `conn.readyState === window.WebSocket.OPEN` was either redundant or perhaps still subject to race conditions.